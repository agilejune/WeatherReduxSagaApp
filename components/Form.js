import React from 'react';
import { View, Text, StyleSheet, TextInput, TouchableOpacity } from 'react-native';

const Form = ({ search, onSetSearch, onSubmit }) => {
    return (
        <View style={styles.container}>
            <View style={styles.heading}>
                <Text style={styles.headingText}>Hello Sunshine!</Text>
            </View>
            <View style={styles.description}>
                <Text style={styles.descriptioinText}>Can you please tell</Text>
                <Text style={styles.descriptioinText}>me the weather</Text>
                <Text style={styles.descriptioinText}>in Germany?</Text>
            </View>
            <View style={styles.content}>
                <Text style={styles.labelText}>Please enter a city</Text>
                <TextInput style={styles.input} value={search} onChangeText={(val) => {onSetSearch(val)}} testID={'input'} />
                <View style={styles.buttonLook}>
                    <TouchableOpacity style={styles.button} onPress={onSubmit} testID={'submit'}>
                        <Text style={styles.buttonText}>Have a look</Text>
                    </TouchableOpacity>
                </View>                
            </View>
        </View>
    );
};

const styles = StyleSheet.create({
    container: {
        paddingTop: 47,
        alignItems: 'center',
    },
    heading: {
        top: 0,
        left: 0,
        width: '100%',
        height: 47,
        alignItems: 'center',
    },
    headingText: {
        textAlign: 'center',
        fontFamily: 'Nunito-Black',
        letterSpacing: 0,
        color: '#F3997B',
        opacity: 1,
        fontSize: 30,
    },
    description: {
        marginTop: 15,
        alignItems: 'center',
        marginBottom: 30,
    },
    descriptioinText: {
        textAlign: 'center',
        fontFamily: 'Nunito-Black',
        letterSpacing: 0,
        opacity: 1,
        color: '#17475B',
        fontSize: 30,
    },
    content: {
        alignItems: 'center',
    },
    labelText: {
        color: '#114358',
        fontFamily: 'Roboto',
        fontSize: 16,
        marginBottom: 14,
    },
    input: {
        height: 45,
        width: 234,
        borderWidth: 2,
        borderColor: '#17475B',
        borderRadius: 1.5,
        marginBottom: 25,
        textAlign: 'center',
        color: '#17475B',
        fontSize: 16,
        fontWeight: 'bold',
        fontFamily: 'Roboto'        
    },
    buttonLook: {
        alignItems: 'center',
    },
    button: {
        backgroundColor: '#F3997B',
        width: 160,
        height: 40,
        justifyContent: 'center',
        borderRadius: 3,
        elevation: 5,
    },
    buttonText: {
        textAlign: 'center',
        fontSize: 12,
        fontFamily: 'Roboto',
        color: 'white',
    },
});

export default Form;