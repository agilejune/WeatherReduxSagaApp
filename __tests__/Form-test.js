import React from 'react';
import Form from '../components/Form';
import {render, cleanup, fireEvent} from 'react-native-testing-library';

afterEach(cleanup);

describe('<Form />', () => {
    it('should match the snapshot', () => {
        const onSetSearch = jest.fn();
        const onSubmit = jest.fn();
        const rendered = render(<Form search={''} onSetSearch={onSetSearch} onSubmit={onSubmit} />).toJSON();
      
        expect(rendered).toMatchSnapshot();
    });

    it('should set search value', () => {
        const onSetSearch = jest.fn();
        const onSubmit = jest.fn();
        const rendered = render(<Form search={'aaa'} onSetSearch={onSetSearch} onSubmit={onSubmit} />);
        const inputComponent = rendered.getByTestId('input');

        expect(inputComponent.props.value).toEqual('aaa');
    });
    
    it('should fire onSetSearch events', () => {
        const onSetSearch = jest.fn();
        const onSubmit = jest.fn();
        const rendered = render(<Form search={''} onSetSearch={onSetSearch} onSubmit={onSubmit} />);
        const inputComponent = rendered.getByTestId('input');
      
        fireEvent(inputComponent, 'changeText', 'new text');
      
        expect(onSetSearch).toHaveBeenCalledWith('new text');
    });
    
    it('should fire onSubmit events', () => {
        const onSetSearch = jest.fn();
        const onSubmit = jest.fn();
        const rendered = render(<Form search={''} onSetSearch={onSetSearch} onSubmit={onSubmit} />);
        const buttonComponent = rendered.getByTestId('submit');
      
        fireEvent(buttonComponent, 'press');
      
        expect(onSubmit).toHaveBeenCalled();
    });
});