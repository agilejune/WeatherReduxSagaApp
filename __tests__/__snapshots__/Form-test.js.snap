// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Form /> should match the snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "paddingTop": 47,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "height": 47,
        "left": 0,
        "top": 0,
        "width": "100%",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#F3997B",
          "fontFamily": "Nunito-Black",
          "fontSize": 30,
          "letterSpacing": 0,
          "opacity": 1,
          "textAlign": "center",
        }
      }
    >
      Hello Sunshine!
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "marginBottom": 30,
        "marginTop": 15,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#17475B",
          "fontFamily": "Nunito-Black",
          "fontSize": 30,
          "letterSpacing": 0,
          "opacity": 1,
          "textAlign": "center",
        }
      }
    >
      Can you please tell
    </Text>
    <Text
      style={
        Object {
          "color": "#17475B",
          "fontFamily": "Nunito-Black",
          "fontSize": 30,
          "letterSpacing": 0,
          "opacity": 1,
          "textAlign": "center",
        }
      }
    >
      me the weather
    </Text>
    <Text
      style={
        Object {
          "color": "#17475B",
          "fontFamily": "Nunito-Black",
          "fontSize": 30,
          "letterSpacing": 0,
          "opacity": 1,
          "textAlign": "center",
        }
      }
    >
      in Germany?
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#114358",
          "fontFamily": "Roboto",
          "fontSize": 16,
          "marginBottom": 14,
        }
      }
    >
      Please enter a city
    </Text>
    <TextInput
      onChangeText={[Function]}
      style={
        Object {
          "borderColor": "#17475B",
          "borderRadius": 1.5,
          "borderWidth": 2,
          "color": "#17475B",
          "fontFamily": "Roboto",
          "fontSize": 16,
          "fontWeight": "bold",
          "height": 45,
          "marginBottom": 25,
          "textAlign": "center",
          "width": 234,
        }
      }
      testID="input"
      value=""
    />
    <View
      style={
        Object {
          "alignItems": "center",
        }
      }
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        nativeID="animatedComponent"
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#F3997B",
            "borderRadius": 3,
            "elevation": 5,
            "height": 40,
            "justifyContent": "center",
            "opacity": 1,
            "width": 160,
          }
        }
        testID="submit"
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontFamily": "Roboto",
              "fontSize": 12,
              "textAlign": "center",
            }
          }
        >
          Have a look
        </Text>
      </View>
    </View>
  </View>
</View>
`;
